src/pydot/dot_parser.py

```diff
@@ -493,7 +513,7 @@ def graph_definition():
     return graphparser
 
 
-def parse_dot_data(s):
+def parse_dot_data(s: str) -> Optional_[List[ParseResults]]:
```

Comment:
Hmm. I wonder if we should change the `except ParseException` block to end with `return []`, and just make this `-> List[ParseResults]`.

...Actually, come to think of it this wouldn't be `List[ParseResults]`, either. It'd be `List[pydot.core.Dot]`. The parser definition might not be annotated properly to _know_ about it — yet? also, this may be a tricky-bordering-on-impossible thing to create proper annotations for — but the result of `graphparser.parse_string()` will always end up being the return value of `push_top_graph_stmt()` if the input is valid and parses to completion.